benchmark/src/test/java/ai/timefold/solver/benchmark/impl/aggregator/BenchmarkAggregatorTest.java

```diff
@@ -0,0 +1,258 @@
+    }
+
+    @Test
+    void aggregateSelectedBenchmarks_directoryIsFile(@TempDir Path tempDir) throws IOException {
+        PlannerBenchmarkConfig config = PlannerBenchmarkConfig
+                .createFromXmlResource("ai/timefold/solver/benchmark/api/testdataBenchmarkConfigWithReport.xml");
+        File regularFile = Files.createFile(tempDir.resolve("regularFile.txt")).toFile();
+        config.setBenchmarkDirectory(regularFile);
+        List<String> selectedDirectories = Arrays.asList("benchmark1", "benchmark2");
+
+        assertThatIllegalArgumentException()
+                .isThrownBy(() -> benchmarkAggregator.aggregateSelectedBenchmarks(config, selectedDirectories))
+                .withMessageContaining("Benchmark directory does not exist");
+    }
+
+    @Test
+    void aggregateSelectedBenchmarks_noValidDirectories() {
+        PlannerBenchmarkConfig config = PlannerBenchmarkConfig
+                .createFromXmlResource("ai/timefold/solver/benchmark/api/testdataBenchmarkConfigWithReport.xml");
+        List<String> nonExistentDirectories = Arrays.asList("nonExistent1", "nonExistent2");
+
+        assertThatIllegalArgumentException()
+                .isThrownBy(() -> benchmarkAggregator.aggregateSelectedBenchmarks(config, nonExistentDirectories))
+                .withMessageContaining("No valid benchmark results found for selected directories");
+    }
+
+    @Test
+    void aggregateSelectedBenchmarks_withValidDirectories() {
+        PlannerBenchmarkConfig config = PlannerBenchmarkConfig
+                .createFromXmlResource("ai/timefold/solver/benchmark/api/testdataBenchmarkConfigWithReport.xml");
+        List<String> availableDirectories = benchmarkAggregator.getAvailableBenchmarkDirectories(config);
+        List<String> selectedDirectories = availableDirectories.subList(0, 2);
+
+        File htmlReportFile = benchmarkAggregator.aggregateSelectedBenchmarks(config, selectedDirectories);
+
+        assertThat(htmlReportFile).isNotNull();
+        assertThat(htmlReportFile.exists()).isTrue();
+        assertThat(htmlReportFile.getName()).endsWith(".html");
+    }
+
+    @Test
+    void getAvailableBenchmarkDirectories_nonExistentDirectory() {
+        PlannerBenchmarkConfig config = new PlannerBenchmarkConfig();
+        config.setBenchmarkDirectory(new File("non/existent/directory"));
+
+        List<String> directories = benchmarkAggregator.getAvailableBenchmarkDirectories(config);
+
+        assertThat(directories).isEmpty();
+    }
+
+    @Test
+    void getAvailableBenchmarkDirectories_emptyDirectory(@TempDir Path tempDir) {
+        PlannerBenchmarkConfig config = new PlannerBenchmarkConfig();
+        config.setBenchmarkDirectory(tempDir.toFile());
+
+        List<String> directories = benchmarkAggregator.getAvailableBenchmarkDirectories(config);
+
+        assertThat(directories).isEmpty();
+    }
+
+    @Test
+    void getAvailableBenchmarkDirectories_withValidDirectories() throws IOException {
```

Comment:
The exception is not thrown, please remove the declaration and apply consistently throughout the file.